### PR TITLE
records: addition of parameterized facets

### PR DESCRIPTION
**Pull request by drjova**
Signed-off-by: Harris Tzovanakis <me@drjova.com>

### CODE REVIEW

**Comment by drjova** [comment]
ci test please

File: inspirehep/modules/search/search_factory.py [hunk 1]
```diff
@@ -69,6 +70,21 @@ def select_source(search, search_index):
     return search
 
 
+def default_inspire_facets_factory(search, index):
+    urlkwargs = MultiDict()
+
+    facets = get_facet_configuration(index)
+
+    if facets is not None:
```
Comment (ammirate) [inline inspirehep/modules/search/search_factory.py]: May `if facets` be more compact?

File: inspirehep/modules/search/facets.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+                },
+            },
+            "author": {
+                "terms": {
+                    "field": "facet_author_name",
+                    "size": 20,
+                    "exclude": exclude_value,
+                },
+                "meta": {
+                    "title": "Collaboartions",
```
Comment (ammirate) [inline inspirehep/modules/search/facets.py]: ops, typo :dart: 
Btw, it should be `Collaborations` or `Collaborators`?

File: tests/integration/search/test_search_facets.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+# or submit itself to any jurisdiction.
+
+from inspirehep.modules.search.facets import hep_author_publications
+
+
+def test_hep_author_publications(isolated_app):
+    expect = {
+        'meta': {
+            'order': 3,
+            'title': 'Collaboartions',
```
Comment (ammirate) [inline tests/integration/search/test_search_facets.py]: :dart:

File: tests/integration/search/test_search_facets.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    with isolated_app.test_request_context('?exclude_author_value=Jones, Jessica'):
+        result = hep_author_publications()
+        assert expect == result['aggs']['author']
+
+
+def test_hep_author_publications_without_exclude_parameter(isolated_app):
+    expect = {
+        'meta': {
+            'order': 3,
+            'title': 'Collaboartions',
```
Comment (ammirate) [inline tests/integration/search/test_search_facets.py]: :dagger:

File: inspirehep/modules/search/facets.py (hunk 2 above)
Comment (michamos) [reply inspirehep/modules/search/facets.py]: `Collaborators`, as those are people.

**Comment by drjova** [comment]
@ammirate updated!

File: setup.py [hunk 5]
```diff
@@ -119,9 +119,9 @@
     'setproctitle~=1.0,>=1.1.10',
     'timeout-decorator~=0.0,>=0.4.0',
     'timeexecution~=3.3.0,<4',
-    # Pin urllib3 to version 1.22 as version 1.23 is incompatible with requirements
-    # from python-requests (<1.23) (https://travis-ci.org/inspirehep/inspire-next/builds/388221674)
-    'urllib3~=1.0,<1.23',
+    # Pin urllib3 to version 1.23 as version 1.24 is incompatible with requirements
+    # from python-requests (<1.24)
+    'urllib3~=1.0,<1.24',
```
Comment (ammirate) [inline setup.py]: I've already seen this :thinking: :smiley:

Comment (drjova) [reply setup.py]: need to rebase :P

**Comment by drjova** [comment]
@ammirate there was an issue with a test only on travis, it's been updated

**Comment by drjova** [comment]
ci merge please

**Comment by drjova** [comment]
ci merge please

**Comment by drjova** [comment]
ci merge please

**Comment by drjova** [comment]
ci merge please

**Comment by drjova** [comment]
ci merge please

**Pull request merged**